TowerForge/TowerForge/LevelManager/Components/GameComponents/DamageComponent.swift

```diff
@@ -48,17 +48,17 @@ class DamageComponent: TFComponent {
             }
 
             // Check collision with opposite team sprite component
-            if oppositeSpriteComponent.node
-                .calculateAccumulatedFrame().intersects(
-                    spriteComponent.node.calculateAccumulatedFrame()) {
-
-                // Check if can attack
-                if CACurrentMediaTime() - lastAttackTime > attackRate {
-                    lastAttackTime = CACurrentMediaTime()
-                    oppositeHealthComponent.decreaseHealth(amount: attackPower)
-                }
-
-            }
+//            if oppositeSpriteComponent.node
```

Comment:
Why is this collision commented?

Reply:
I changed the collision detection to a general one so that instead of all components needing to check collisions with all other components, we can check collisions between entities instead which can call the appropriate components to check and generate the appropriate events to be resolved by the systems.

Reply:
yup sure that's great